spec/models/claim/matching_attribute_finder_spec.rb

```
@@ -0,0 +1,83 @@
+    let(:source_claim) {
+      create(
+        :claim,
+        teacher_reference_number: "0902344",
+        national_insurance_number: "QQ891011C",
+        email_address: "genghis.khan@mongol-empire.com",
+        bank_account_number: "34682151",
+        bank_sort_code: "972654",
+        building_society_roll_number: "123456789/ABCD",
+      )
+    }
+
+    let!(:claim_with_no_matching_attributes) { create(:claim, :submitted) }
+    let!(:unsubmitted_claim_with_matching_teacher_reference_number) { create(:claim, :submittable, teacher_reference_number: source_claim.teacher_reference_number) }
+
+    let(:claim_with_matching_attribute) { build(:claim, :submitted) }
+    let(:claims_with_matching_attributes) { Claim::MatchingAttributeFinder.new(source_claim).claim_ids_with_matching_attribute_keys }
+
+    it "does not include the id of the source claim" do
+      expect(claims_with_matching_attributes.keys).not_to include(source_claim.id)
+    end
+
+    it "does not include the id of a claim without any matcheing attributes" do
+      expect(claims_with_matching_attributes.keys).not_to include(claim_with_no_matching_attributes.id)
+    end
+
+    it "only includes submitted claims" do
+      expect(claims_with_matching_attributes.keys).not_to include(unsubmitted_claim_with_matching_teacher_reference_number.id)
+    end
+
+    it "includes the id of a claim with a mathching teacher reference number" do
+      claim_with_matching_attribute.teacher_reference_number = source_claim.teacher_reference_number
+      claim_with_matching_attribute.save
+
+      expect(claims_with_matching_attributes.keys).to include(claim_with_matching_attribute.id)
+      expect(claims_with_matching_attributes[claim_with_matching_attribute.id]).to include("Teacher reference number")
+    end
+
+    it "includes the id of a claim with a mathching national insurance number" do
+      claim_with_matching_attribute.national_insurance_number = source_claim.national_insurance_number
```

You could save a line, losing the `save`, with

```suggestion
      claim_with_matching_attribute.update(national_insurance_number: source_claim.national_insurance_number)
```

And it might be worth using `update!` (or `save!`) just to make sure it fails loudly if the test setup fails.

This applies to the other similar `it` blocks in this file, too.

---

Much nicer with `update!` thank you.